src/compiler/config/outputs/validate-dist.ts

```diff
@@ -102,49 +110,48 @@ export const validateDist = (config: d.Config, userOutputs: d.OutputTarget[]) =>
   }, []);
 };
 
-const validateOutputTargetDist = (config: d.Config, o: d.OutputTargetDist) => {
+/**
+ * Validate that an OutputTargetDist object has what it needs to do it's job.
+ * To enforce this, we have this function return
+ * `Required<d.OutputTargetDist>`, giving us a compile-time check that all
+ * properties are defined (with either user-supplied or default values).
+ *
+ * @param config the current config
+ * @param o the OutputTargetDist object we want to validate
+ * @returns `Required<d.OutputTargetDist>`, i.e. `d.OutputTargetDist` with all
+ * optional properties rendered un-optional.
+ */
+const validateOutputTargetDist = (config: d.Config, o: d.OutputTargetDist): Required<d.OutputTargetDist> => {
+  // we need to create an object with a bunch of default values here so that
+  // the typescript compiler can infer their types correctly
   const outputTarget = {
     ...o,
     dir: getAbsolutePath(config, o.dir || DEFAULT_DIR),
+    buildDir: isString(o.buildDir) ? o.buildDir : DEFAULT_BUILD_DIR,
+    collectionDir: isString(o.collectionDir) ? o.collectionDir : DEFAULT_COLLECTION_DIR,
```

Comment:
The original behavior for `collectionDir` checked for strict equality against `undefined`, can we restore that check?

[Original](https://github.com/ionic-team/stencil/pull/3333/files#diff-1677525851f8c0d25cf1838e531ac1778ed57f1bba3cf62a86923e451c47904bL119):
```ts
  if (outputTarget.collectionDir === undefined) {
    outputTarget.collectionDir = DEFAULT_COLLECTION_DIR;
  }
```

Suggested:
```suggestion
    collectionDir: o.collectionDir !== undefined ? o.collectionDir : DEFAULT_COLLECTION_DIR,
```

Reply:
yeah that makes sense, I'll make that change real quick

Reply:
made that change in https://github.com/ionic-team/stencil/pull/3333/commits/a22e380d2b8a0ef4fca3371bf5c69fe4d088ccb4